app/scripts/directives/editKeyValueMap.js

```diff
@@ -1,16 +1,17 @@
 "use strict";
 
 angular.module("openshiftConsole")
-  .directive("editConfigMap",
+  .directive("editKeyValueMap",
```

Comment:
I worry that this name is too close to `key-value-editor` and will cause confusion.

https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/keyValueEditor.js#L6

Maybe make it clear that it is specific to config maps and secrets? `editConfigMapOrSecret` too on the nose?

Reply:
How about `key-value-uploader`, since it does add/remove key-value-pairs, but also allows `value` to be uploaded from a file.

Reply:
It's pretty specific to config maps and secrets rather than something general. For instance, we perform validation of keys using specific regex and maxlength checks specific to those resources. I still feel like it would be good to convey that in the name.